vaadin-button.html

```diff
@@ -22,25 +22,187 @@
+             */
+            disabled: {
+              type: Boolean,
+              default: false,
+              reflectToAttribute: true,
+              observer: '_disabledChanged',
+            },
+
+            /**
+             * role="button"
+             */
+            role: {
+              type: String,
+              value: 'button',
+              reflectToAttribute: true,
+              readOnly: true
+            },
+
+            /**
+             * The position of the element in the tabbing navigation order for the current document.
+             */
+            tabindex: {
+              type: Number,
+              value: 0,
+              reflectToAttribute: true
+            },
+
+            /**
+             *
+             */
+            _previousTabIndex: Number
           }
         };
       }
-    }
 
-    customElements.define(VaadinElement.is, VaadinElement);
+      ready() {
+        this._addListeners();
+
+        super.ready();
+
+        if (this.autofocus) {
+          window.requestAnimationFrame(function() {
+            this.focus();
+          }.bind(this));
+        }
+      }
+
+      activate(e) {
+        this.active = true;
+        Polymer.Base.fire('click', e, {
+          bubbles: false,
+          node: this
+        });
+      }
+
+      deactivate(e) {
+        this.active = false;
+      }
+
+      _disabledChanged(disabled) {
+        if (disabled) {
+          this._previousTabIndex = this.tabindex;
+          this.tabindex = -1;
+          this._removeListeners();
+        } else {
+          this.tabindex = this._previousTabIndex;
+          this._addListeners();
+        }
+      }
+
+      _addListeners() {
+        this.addEventListener('mousedown', this._handleMouseDown.bind(this));
+        this.addEventListener('touchstart', this._handleMouseDown.bind(this));
+        this.addEventListener('mouseup', this._handleMouseUp.bind(this));
+        this.addEventListener('touchend', this._handleMouseUp.bind(this));
+        this.addEventListener('keydown', this._handleKeyDown.bind(this));
+        this.addEventListener('keyup', this._handleKeyUp.bind(this));
+      }
+
+      _removeListeners() {
+        this.removeEventListener('mousedown', this._handleMouseDown);
+        this.removeEventListener('touchstart', this._handleMouseDown);
+        this.removeEventListener('mouseup', this._handleMouseUp);
+        this.removeEventListener('touchend', this._handleMouseUp);
+        this.removeEventListener('keydown', this._handleKeyDown);
+        this.removeEventListener('keyup', this._handleKeyUp);
+      }
+
+      _handleMouseDown(e) {
+        if (e.which === 1) { // left button
+          this.activate(e);
+        }
+      }
+
+      _handleMouseUp(e) {
+        if (e.which === 1) { // left button
+          this.deactivate(e);
+        }
+      }
+
+      _handleKeyDown(e) {
+        if (e.keyCode === 13 || e.keyCode === 32) {
```

Comment:
Would be helpful to add comment here to say which keys the keycodes are. Alternatively, we could look into using `iron-a11y-keys` or `iron-a11y-keys-behavior` if it's available for Polymer 2.0 – at least the static function `Polymer.IronA11yKeysBehavior. keyboardEventMatchesKeys(event, 'enter')` should be available. Might be just a matter of taste if it's worth adding a new dependency.